### ISSUE TITLE
Build failed while deploying to Heroku, Can't resolve 'core-js/es7/reflect'

### ISSUE THREAD

**Issue by freewebwithme** [juice-shop/juice-shop#1815]
```      
 ./src/polyfills.ts:47:0-29 - Error: Module not found: Error: Can't resolve 'core-js/es7/reflect' in '/tmp/build_8197b157/frontend/src'
       npm ERR! code 1
       npm ERR! path /tmp/build_8197b157
       npm ERR! command failed
       npm ERR! command sh -c cd frontend && npm install --legacy-peer-deps && cd .. && npm run build:frontend && (npm run --silent build:server || cd .)
       npm ERR! A complete log of this run can be found in:
       npm ERR!     /tmp/npmcache.XEDKa/_logs/2022-05-21T23_13_21_269Z-debug-0.log

-----> Build failed
```

**Comment by jiachen247** [comment]
Yes can confirm this is happening to me too, it worked a few days back (17 May) but now when I do a clean install it shows this error. Tested both on Heroku and locally.

More detailed logs:
```
> juice-shop@13.4.0-SNAPSHOT build:frontend /Users/jiachen/juice-shop
> cd frontend && node ./node_modules/@angular/cli/bin/ng build --aot --configuration production --output-hashing=none --vendor-chunk=true --source-map=false

✔ Browser application bundle generation complete.

./src/polyfills.ts:47:0-29 - Error: Module not found: Error: Can't resolve 'core-js/es7/reflect' in '/Users/jiachen/juice-shop/frontend/src'


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! juice-shop@13.4.0-SNAPSHOT build:frontend: `cd frontend && node ./node_modules/@angular/cli/bin/ng build --aot --configuration production --output-hashing=none --vendor-chunk=true --source-map=false`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the juice-shop@13.4.0-SNAPSHOT build:frontend script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jiachen/.npm/_logs/2022-05-22T06_45_37_198Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! juice-shop@13.4.0-SNAPSHOT postinstall: `cd frontend && npm install --legacy-peer-deps && cd .. && npm run build:frontend && (npm run --silent build:server || cd .)`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the juice-shop@13.4.0-SNAPSHOT postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jiachen/.npm/_logs/2022-05-22T06_45_37_291Z-debug.log
```

**Comment by bkimminich** [comment]
As our last CI/CD jobs passed fine, including Heroku deployment, I assume that this is some upstream dependency problem that we now pull via some of our unpinned `~` or `^` dependencies.

I won't be able to do much investigation next week unfortunately, so volunteers are very welcome to pick this up.

**Comment by bkimminich** [comment]
@jiachen247 For easier local reproduction, please state your Node.js version, OS etc. and please check if Juice Shop 14.x works for you (please check `master` branch _and_ pre packaged ZIP of 14.x - if it is a dependency problem, the ZIP should work for you, as it already contains als deps and doesn't need an install step that could break those)

**Comment by jiachen247** [comment]
Sadly I'm using and M1 (arm64) which is not yet a compilation target for juice shop project! Thanks for all the help, ill just deploy a bunch VMs via docker for now! :-)

Here's my setup, have also tried running on ubuntu and also the one click Heroku deploy that also runs into the same problem.

```
(base) jiachen@jiachens-MacBook-Pro-423 ~ % uname -a
Darwin jiachens-MacBook-Pro-423.local 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64
(base) jiachen@jiachens-MacBook-Pro-423 ~ % node -v
v14.18.2
```

I have a version locally thats working (installed May 17) but there seems to be no lock file generated to compare against. Would love to help if you have any ideas (short of pinning the dep one by one)!

**Comment by bkimminich** [comment]
Problem is fixed and I'll release a hotfix `v14.0.1` now for it. I don't understand how/why this occured _now_, but it seems a path in our `polyfill.ts` no longer exists. We expected `core-js/es7/reflect` but it _now_ is `core-js/es/reflect`.
